[DataLoader] Rename Functional DataSet to DataPipe

ejguan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51488 [DataLoader] Rename Functional DataSet to DataPipe**
* #51126 [WIP][DataLoader] Implement BucketBatchIterableDataset
* #50045 [WIP][DataLoader] Implement CallableIterableDataset

Differential Revision: [D26209888](https://our.internmc.facebook.com/intern/diff/D26209888)

ejguan committed:
[DataLoader] Rename Functional DataSet to DataPipe

[ghstack-poisoned]

glaringlee commented on torch/utils/data/datapipes/iter/batch.py:
```diff
@@ -1,42 +1,42 @@
         assert batch_size > 0, "Batch size is required to be larger than 0!"
-        super(BatchIterableDataset, self).__init__()
-        self.dataset = dataset
+        super(BatchIterDataPipe, self).__init__()
```
nit: I think super().__init__() should work in python 3.2 and later?

glaringlee commented on torch/utils/data/datapipes/iter/batch.py:
```diff
@@ -49,60 +49,60 @@ def __iter__(self) -> Iterator[List[T_co]]:
         assert batch_size > 0, "Batch size is required to be larger than 0!"
-        super(BucketBatchIterableDataset, self).__init__()
-        self.dataset = dataset
+        super(BucketBatchIterDataPipe, self).__init__()
```
same here.

glaringlee commented on torch/utils/data/datapipes/iter/callable.py:
```diff
@@ -1,36 +1,36 @@
                  ) -> None:
-        super(CallableIterableDataset, self).__init__()
-        self.dataset = dataset
+        super(CallableIterDataPipe, self).__init__()
```
Same here

glaringlee commented on torch/utils/data/datapipes/iter/callable.py:
```diff
@@ -72,21 +72,21 @@ class CollateIterableDataset(CallableIterableDataset):
                  **kwargs,
                  ) -> None:
-        super(CollateIterableDataset, self).__init__(dataset, *args, fn=collate_fn, **kwargs)
+        super(CollateIterDataPipe, self).__init__(datapipe, *args, fn=collate_fn, **kwargs)
```
Same here.

glaringlee commented on torch/utils/data/datapipes/iter/sampler.py:
```diff
@@ -1,32 +1,32 @@
-        self.dataset = dataset
+        assert isinstance(datapipe, Sized), \
+            "Sampler class requires input datapipe implemented `__len__`"
+        self.datapipe = datapipe
```
Are we missing super().__init__()? Though it seems fine for now.

glaringlee reviewed (approved):
LGTM, some nit comments, please update when rebasing the code.

ejguan committed:
Update on "[DataLoader] Rename Functional DataSet to DataPipe"




[ghstack-poisoned]

ejguan committed:
Update on "[DataLoader] Rename Functional DataSet to DataPipe"


Differential Revision: [D26209888](https://our.internmc.facebook.com/intern/diff/D26209888)

[ghstack-poisoned]
